sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/HostResource.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.resources;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+/** Factory of a {@link Resource} which provides information about the host info. */
+public final class HostResource {
+
+    private static final Resource INSTANCE = buildResource();
+
+    /**
+     * Returns a factory for a {@link Resource} which provides information about host.
+     */
+    public static Resource get() {
+        return INSTANCE;
+    }
+
+    // Visible for testing
+    static Resource buildResource() {
```

Comment:
Do you think we can add the `host.arch` attribute too?

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/host.md#host

Reply:
I think is a good idea.I will add it.